libcontainer/process_linux.go

```diff
@@ -83,8 +83,7 @@ func (p *setnsProcess) start() (err error) {
 	if err = p.execSetns(); err != nil {
 		return newSystemErrorWithCause(err, "executing setns process")
 	}
-	// We can't join cgroups if we're in a rootless container.
-	if !p.config.Rootless && len(p.cgroupPaths) > 0 {
```

Comment:
We're not clear under what circumstances the rootless check still makes sense? Can you give an example please?

`p.cgroupPaths` is loaded from the `state.json` which best we can tell has to be the same as the cgroup paths the init process is in, so barring people doing weird things, if you succeeded on create, you should succeed now?